slam_toolbox/src/loop_closure_assistant.cpp

```diff
@@ -125,48 +125,93 @@ void LoopClosureAssistant::publishGraph()
 /*****************************************************************************/
 {
   interactive_server_->clear();
-  std::unordered_map<int, Eigen::Vector3d>* graph = solver_->getGraph();
+  karto::MapperGraph * graph = mapper_->GetGraph();
 
-  if (graph->size() == 0)
+  if (!graph || graph->GetVertices().empty())
   {
     return;
   }
 
-  ROS_DEBUG("Graph size: %i",(int)graph->size());
   bool interactive_mode = false;
   {
     boost::mutex::scoped_lock lock(interactive_mutex_);
     interactive_mode = interactive_mode_;
   }
 
-  visualization_msgs::MarkerArray marray;
-  visualization_msgs::Marker m = vis_utils::toMarker(map_frame_,
-    "slam_toolbox", 0.1);
+  // schedule all existing markers for removal
+  for (auto it = markers_.begin(); it != markers_.end(); ++it) {
+    it->second.action = visualization_msgs::Marker::DELETE;
+  }
+
+  visualization_msgs::Marker vertex_marker =
+    vis_utils::toVertexMarker(map_frame_, "slam_toolbox", 0.1);
 
-  for (ConstGraphIterator it = graph->begin(); it != graph->end(); ++it)
+  for (const auto & vertices_per_sensor : graph->GetVertices())
```

Comment:
Please use the old style of for loops to be consistent in codebase

Reply:
Done in https://github.com/SteveMacenski/slam_toolbox/pull/467/commits/9f4eb994925ce5246694f4d76656802bba562ef5.